src/calculateRentalCost.js

```diff
@@ -4,7 +4,23 @@
  * @return {number}
  */
 function calculateRentalCost(days) {
-  // write code here
+  const DAILY_RENTAL_COST = 40;
+  const LONG_TERM_THRESHOLD = 7;
+  const LONG_TERM_DISCOUNT = 50;
+  const MID_TERM_THRESHOLD = 3;
+  const MID_TERM_DISCOUNT = 20;
+
+  const totalCost = days * DAILY_RENTAL_COST;
+
+  if (days >= LONG_TERM_THRESHOLD) {
+    return totalCost - LONG_TERM_DISCOUNT;
+  }
+
+  if (days >= MID_TERM_THRESHOLD) {
+    return totalCost - MID_TERM_DISCOUNT;
```

Comment:
There is a logical issue in the discount application. The current implementation applies the long-term discount for rentals of 7 days or more, and the mid-term discount for rentals of 3 to 6 days. However, if a rental is exactly 7 days, it should only apply the long-term discount, not both. The logic should be adjusted to ensure that only one discount is applied based on the highest applicable threshold.